closed/src/java.base/share/classes/com/sun/crypto/provider/NativeCipherBlockChaining.java

```diff
@@ -217,7 +223,10 @@ void save() {
      */
     void restore() {
         System.arraycopy(rSave, 0, r, 0, blockSize);
-        int ret = nativeCrypto.CBCInit(nativeContext, mode, r, r.length, key, key.length);
+        int ret;
+        synchronized (this) {
+            ret = nativeCrypto.CBCInit(nativeContext, mode, r, r.length, key, key.length);
+        }
```

Comment:
In jdk16, synchronization was added `encrypt()`, but not in this version: why not?